lib/matplotlib/cbook/__init__.py

```diff
@@ -199,48 +199,61 @@ def __setstate__(self, state):
 
     def connect(self, s, func):
         """Register *func* to be called when signal *s* is generated."""
-        self._func_cid_map.setdefault(s, {})
+        signal = s
```

Comment:
please revert this rename (or perhaps we could consider `@api._rename_parameter`ing "s" to "signal" if you really want to improve that.)

Reply:
Thanks, I will check your decorator. I definitely prefer renaming it. It's better for everybody, including people using this API.

Reply:
I have just added the rename param